coderd/inboxnotifications.go

```diff
@@ -0,0 +1,376 @@
+package coderd
+
+import (
+	"context"
+	"database/sql"
+	"encoding/json"
+	"net/http"
+	"slices"
+	"strings"
+	"time"
+
+	"github.com/go-chi/chi/v5"
+	"github.com/google/uuid"
+	"golang.org/x/xerrors"
+
+	"cdr.dev/slog"
+
+	"github.com/coder/coder/v2/coderd/database"
+	"github.com/coder/coder/v2/coderd/database/dbtime"
+	"github.com/coder/coder/v2/coderd/httpapi"
+	"github.com/coder/coder/v2/coderd/httpmw"
+	"github.com/coder/coder/v2/coderd/pubsub"
+	"github.com/coder/coder/v2/codersdk"
+	"github.com/coder/coder/v2/codersdk/wsjson"
+	"github.com/coder/websocket"
+)
+
+// convertInboxNotificationParameters parses and validates the common parameters used in get and list endpoints for inbox notifications
+func convertInboxNotificationParameters(ctx context.Context, logger slog.Logger, targetsParam string, templatesParam string, readStatusParam string) ([]uuid.UUID, []uuid.UUID, string, error) {
```

Comment:
It is a general comment - if we want to use inbox just for notifications, we should adjust filenames and function names to indicate the relationship: `notificationsinbox.go` or `convertNotificationsInboxParameters`.